File: internal/domain/application.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Application struct {
	ID          uuid.UUID `json:"id"`
	Name        string    `json:"name"`
	Slug        string    `json:"slug"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type Environment struct {
	ID               uuid.UUID `json:"id"`
	ApplicationID    uuid.UUID `json:"application_id"`
	Name             string    `json:"name"`
	Slug             string    `json:"slug"`
	RequiresApproval bool      `json:"requires_approval"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
}

File: src/main/java/com/ecommerce/featuremanagement/domain/Application.java
package com.ecommerce.featuremanagement.domain;

import java.time.Instant;
import java.util.UUID;

public class Application {
    public UUID id;
    public String name;
    public String slug;
    public String description;
    public Instant createdAt;
    public Instant updatedAt;
}
